awpy/analytics/map_control.py

```diff
@@ -1,330 +1,368 @@
+                map_control_values[cur_id].append(cur_tile["map_control_value"])
+
+                neighbors = [TileIdObject(i) for i in neighbor_info[cur_id]]
                 if len(neighbors) == 0:
                     neighbors = [
-                        tile_id for _, tile_id in approx_neighbors(map_name, cur_id)
+                        tile["tile_id"] for tile in _approx_neighbors(map_name, cur_id)
                     ]
-                for neighbor in neighbors:
-                    stack.append((neighbor, cur_val - step_size))
 
-    return current_dict
+                queue.extend(
+                    [
+                        BFSTileData(
+                            {
+                                "tile_id": neighbor,
+                                "map_control_value": (cur_tile["steps_left"] - 1)
+                                / max_depth,
+                                "steps_left": cur_tile["steps_left"] - 1,
+                            }
+                        )
+                        for neighbor in neighbors
+                    ]
+                )
+
+    return map_control_values
 
 
-def frame_tile_map_control_values(
+def _frame_tile_map_control_values(
     map_name: str,
-    ct_tiles_wanted: list[int],
-    t_tiles_wanted: list[int],
-    neighbor_info: dict[int, set[int]],
-) -> tuple[dict[int, list[float]], dict[int, list[float]]]:
+    ct_tiles: list[TileIdObject],
+    t_tiles: list[TileIdObject],
+    neighbor_info: TileNeighborsDict,
+) -> FrameMapControlValues:
     """Calculate a frame's map control values for each side.
 
     Args:
-        map_name (string)        : Map for other arguments
-        ct_tiles_wanted (list)   : List of tiles where ct players
-                                   are located
-        t_tiles_wanted (list)    : List of tiles where t players
-                                   are located
-        neighbor_info (dict)     : Dictionary mapping tile to its
-                                   navigable neighbors
+        map_name (str): Map for other arguments
+        ct_tiles (list): List of CT-occupied tiles
+        t_tiles (list): List of T-occupied tiles
+        neighbor_info (TileNeighborsDict): Object with tile to neighbor mapping
 
     Returns:
-        Tuple containing map control dictionary for each side
-            Tuple[0] -> CT Dict
-            Tuple[1] -> T Dict
-    Raises:
-        ValueError: If map_name is not in awpy.data.NAV
+        FrameMapControlValues object containing each team's map control values
+
+    Values are allocated to tiles depending on how many tiles are between it
+    and the source tile (aka tile distance). The smaller the tile distance,
+    the close the tile's value is to 1. The tile distance can get as high as
+    the max_depth argument, which is 10 as default. If the max_depth is kept
+    as default 10, this means the BFS search will stop once a tile
+    distance of 10 is reached.
```

Comment:
Same here with the being before the Args.

Reply:
Addressed in the next commit.